client/src/components/Heroes/HeroesRadarTab.tsx

```diff
@@ -0,0 +1,99 @@
+import { Button, Checkbox, Form, Select, Space, TableProps } from 'antd';
+import HeroesRadarTable from './HeroesRadarTable';
+import { HeroesRadarDto, GratitudesApi, HeroRadarDto } from 'api';
+import { IPaginationInfo } from 'common/types/pagination';
+import { useState, useEffect, useCallback } from 'react';
+import { Course } from 'services/models';
+import { onlyDefined } from 'utils/onlyDefined';
+
+export type HeroesRadarFormProps = {
+  courseId?: number;
+  notActivist?: boolean;
+};
+
+type GetHeroesProps = HeroesRadarFormProps & Partial<IPaginationInfo>;
+
+export type LayoutType = Parameters<typeof Form>[0]['layout'];
+
+const initialPage = 1;
+const initialPageSize = 20;
+const initialQueryParams = { current: initialPage, pageSize: initialPageSize };
+
+function HeroesRadarTab({ setLoading, courses }: { setLoading: (arg: boolean) => void; courses: Course[] }) {
+  const [heroes, setHeroes] = useState<HeroesRadarDto>({
+    content: [],
+    pagination: { current: initialPage, pageSize: initialPageSize, itemCount: 0, total: 0, totalPages: 0 },
+  });
+  const [form] = Form.useForm();
+  const [formData, setFormData] = useState<HeroesRadarFormProps>(form.getFieldsValue());
+  const [formLayout, setFormLayout] = useState<LayoutType>('inline');
+  const gratitudeApi = new GratitudesApi();
+
+  const getHeroes = async ({
+    current = initialPage,
+    pageSize = initialPageSize,
+    courseId,
+    notActivist,
+  }: GetHeroesProps) => {
+    try {
+      setLoading(true);
+      const { data } = await gratitudeApi.getHeroesRadar(current, pageSize, courseId, notActivist);
+      setHeroes(data);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    getHeroes(initialQueryParams);
+  }, []);
+
+  const makeRequest = useCallback(async (data: HeroesRadarFormProps) => {
```

Comment:
it seems redundant. You re-create `getHeroes` on every render but cache `makeRequest`. Not sure if it has benefits

Reply:
@apalchys is it better now?